### ISSUE TITLE
[Radio/Checkbox] disabled state makes label illegible

### ISSUE THREAD

**Issue by theetrain** [telus/tds-core#1294]
## Description

Originally reported by @catherinetelus

The core-radio and core-checkbox components both allow `disabled` state in DSM and React, though not explicitly documented. When disabled, the label text greys out, making it difficult to read.

## Reproduction Steps

1. Apply the `disabled` prop to core-radio or core-checkbox
2. Observe label text rendering in Gainsboro Grey

## Workaround details

No available workarounds.

## Recommendation

- Render label text in black, despite disabled state
- Ensure elements that are both checked and disabled render in grey, not green
- Document built in accessibility functionality, and any applicable accessibility guidelines

## Meta

- TDS component version: `@tds/core-radio`, `@tds/core-checkbox`
- Willing to develop a solution: ?
- High impact: No 

## Screenshots

![image](https://user-images.githubusercontent.com/12798751/67396378-7c7c2400-f575-11e9-9e06-b4f8ec570325.png)
![image](https://user-images.githubusercontent.com/12798751/67396418-8f8ef400-f575-11e9-9e67-e1bceeb5c1a3.png)

**Comment by theetrain** [comment]
Next steps:

Confirm with Accessibility office hours.

**Comment by theetrain** [comment]
Feedback from @mejsullivan:

* Take back to TDS: possibly make the label black when disabled, change green visual indicators to grey; but that decision is ours to make. Try auditing other use cases and come up with a proposal for CX. See if we can make a checklist to help decide whether disabled radio buttons are right for you.
* Docs idea: Avoid having disabled radio buttons at all, but rather only show available options and indicate unavailable items in a different way (tooltip? expandcollapse? other?). When an item is disabled, you should assume its information is completely irrelevant or unimportant to the customer.
* Use this opportunity to look into high contrast mode on Windows, to see how the radio and checkbox indicators appear, along with their labels in active, inactive, and disabled states

**Comment by kittybourne** [comment]
Another option is to separate the list of available stores from the non available stores. With only the available stores selectable with radio button. 

@catherinetelus

**Comment by theetrain** [comment]
Moving forward, we may go with the following:

* Black text for disabled radios/checkboxes
* Grey selectable artifacts (should not be green)

**Comment by catherinetelus** [comment]
> Another option is to separate the list of available stores from the non available stores. With only the available stores selectable with radio button.
> 
> @catherinetelus

@kittybourne I think we will want to list the stores in sequence of proximity first and foremost, but we will implement a filter by default that only show radio button list of available stores. But once unchecked, the list would show all stores (selectable/available and unselectable/unavailable).

**Comment by theetrain** [comment]
Based on conversations we've had with CX and Accessibility, the TDS team will take action on the guidelines and appearance of the Radio and Checkbox components based on the following considerations:

* Currently, the disabled Radio component shows a green dot when active when it should be grey
* Guidelines should be provided around the decision to disable radios or checkboxes in general. It's not recommended to display disabled radios or checkboxes within fieldsets that have active checkboxes and radios since screen magnifier, keyboard, and screen reader users will likely skip over them completely
* We could display the labels of disabled components with high-contrast text, but as per the previous point, some users may as well consider the copy as nonexistent. It's safe to keep the low-contrast text and treat its content as non-existent for all users

**Comment by catherinetelus** [comment]
@theetrain would this differ from browser defaults? Also, I thought that disabled labels were still read but the status "dimmed" was appended?

**Comment by nadaltelus** [comment]
Closing the issue due to lack of activity. If surfaced again, will consider as input for the new DS.
